File: backend/app/models/game.py
from pydantic import BaseModel
from typing import List, Optional
from enum import Enum

class Direction(str, Enum):
    UP = "UP"
    DOWN = "DOWN"

class GameRound(BaseModel):
    id: str
    chartData: List[float]
    hiddenData: List[float]
    correctAnswer: Direction
    timestamp: float
    pair: str = "ETH-USDT"

class User(BaseModel):
    id: str
    balance: int
    gamesPlayed: int
    gamesWon: int
    totalWinnings: int

class GameSession(BaseModel):
    sessionId: str
    userId: str
    roundId: str
    bet: int
    userAnswer: Optional[Direction] = None
    responseTime: Optional[float] = None
    isCorrect: Optional[bool] = None
    payout: Optional[int] = None
    timestamp: float

class NewGameResponse(BaseModel):
    roundId: str
    chartData: List[float]
    pair: str
    timestamp: float

class AnswerRequest(BaseModel):
    roundId: str
    userAnswer: Direction
    bet: int
    responseTime: float

class AnswerResponse(BaseModel):
    isCorrect: bool
    correctAnswer: Direction
    payout: int
    newBalance: int
    coefficient: float 

File: frontend/src/types/game.ts
export enum Direction {
  UP = "UP",
  DOWN = "DOWN",
}

export interface GameRound {
  roundId: string;
  chartData: number[];
  pair: string;
  timestamp: number;
}

export interface AnswerRequest {
  roundId: string;
  userAnswer: Direction;
  bet: number;
  responseTime: number;
}

export interface AnswerResponse {
  isCorrect: boolean;
  correctAnswer: Direction;
  payout: number;
  newBalance: number;
  coefficient: number;
}

export interface User {
  id: string;
  balance: number;
  gamesPlayed: number;
  gamesWon: number;
  totalWinnings: number;
}

export interface GameState {
  currentRound: GameRound | null;
  userBalance: number;
  selectedBet: number;
  isGameActive: boolean;
  timer: number;
  lastResult: AnswerResponse | null;
}

export type BetAmount = 10 | 50 | 100;
